googleSchema.js

```diff
@@ -0,0 +1,55 @@
+// Philosophy FAQ
+// to add more philosophy questions just add them in this json as the first 2 and the script will do the rest
+var faqPhilosophy = {
+	"faqs": {
+		"protectingOurDemographics": {
+			"question": "What does Destiny think about protecting our demographics?",
+			"answer": "Destiny believes that this endeavor is futile because he rejects the notion that there is a consistent and coherent definition of American values",
+		},
+		"protestingRioting": {
+			"question": "What does Destiny think about the protesting and rioting?",
+			"answer": "Destiny ardently believes in people's right to protest and will always support the right to protest, even if he doesn't necessarily agree with the positions being advocated for by any particular protest. ",
+		},
+	}
+};
+
+// Initialize JSON array and data
+var faqsPhilosophy = faqPhilosophy.faqs;
+faqsPhilosophy = JSON.stringify(faqsPhilosophy);
+faqsPhilosophy = JSON.parse(faqsPhilosophy);
+
+var string = `{"@context": "https://schema.org","@type": "FAQPage","mainEntity":[`;
+
+var iterateString = '';
+// iterates over the questions
+Object.values(faqsPhilosophy).filter(faqsPhilosophy =>
+	iterateString += `{"@type": "Question","name": "` + faqsPhilosophy.question + `","acceptedAnswer": {"@type": "Answer","text": "` + faqsPhilosophy.answer + `"}},`
+);
+
+//slices last ','
+iterateString = iterateString.slice(0, -1);
+
+string += iterateString;
+string += `]}`;
+
+// inject string into markdown
+if (typeof window !== 'undefined') {
+  window.onload = function(){
+		const elementExists = document.getElementById("philosophy-faq");
+		if (elementExists) {
+			document.querySelector("#philosophy-faq").innerHTML = string;
+		}
+	};
+	//TODO add recurring interval when switching location href, find a better way than calling an interval every 500 ms
+	var currentPage = location.href;
+	setInterval(function()
```

Comment:
Don't use an interval, use a listener like `locationChange`

Reply:
I can't seem to get the listener to work since it's an SPA. Any additional help here is welcomed
I tried using pushstate but it doesn't detect the element when the page url was changed, but rather saves it and shows it when the page is changed again ...?

```
//checks if the client has changed link
let schemaPushState = history.pushState;
history.pushState = function () {
	schemaPushState.apply(history, arguments);
	const elementExists = document.getElementById("philosophy-faq");
	if (elementExists) {
		document.querySelector("#philosophy-faq").innerHTML = schemaString;
	}
};
```

Reply:
I know Docusaurus lets you wrap theme components.
You could try and wrap the component that contains each page and use react hooks or lifecycle functions to execute your JS. Either `useEffect(() => <your js code that runs on page load>, [])` or `componentDidMount(){<your js code>}`

This describes how to wrap a theme component without rewriting it
https://v2.docusaurus.io/docs/using-themes#wrapping-theme-components

Reply:
https://github.com/destinygg/destinypositions/commit/8c0f5456274ed0c5ee65ec0c2fdd0e4dee67c03a
Here's an example that just console log's "test" on each document load.

Reply:
works flawlessly, thank you so much @mattroseman!